bagua/torch_api/communication.py

```diff
@@ -68,7 +70,16 @@ class ReduceOp(IntEnum):
 @gorilla.patches(TorchProcessGroup, filter=lambda name, obj: "bagua" in name)
 class BaguaProcessGroupPatch:
     def bagua_patch(self, stream: Optional[torch.cuda.Stream] = None):
-        self.bagua_pg = from_torch_group(self, stream)
+        global _torch_to_bagua_pg_map
+        if self not in _torch_to_bagua_pg_map:
+            _torch_to_bagua_pg_map[self] = from_torch_group(self, stream)
+
+        return self
+
+    @property
+    def bagua_pg(self):
+        assert self in _torch_to_bagua_pg_map, "The torch process group not in cache, you should execute class method `bagua_patch` first"
```

Comment:
"cannot find associated Bagua process group in cache, BaguaProcessGroupPatch.bagua_patch(...) needs to be run first to initialize Bagua process group in cache."